opm/simulators/wells/WellGroupHelpers.cpp

```diff
@@ -760,8 +760,11 @@ namespace WellGroupHelpers
                 throw std::invalid_argument("Invalid Flow target type in GPMAINT");
         }
         auto& gpmaint_state = group_state.gpmaint(group.name());
-        double rate = gpm->rate(gpmaint_state, current_rate, error, dt);
-        group_state.update_gpmaint_target(group.name(), std::max(0.0, sign * rate));
+        // we only activate gpmaint if pressure is lower than the target regional pressure for injectors
+        // (i.e. error > 0) and higher for producers.
+        bool activate = (injection && error > 0) || (!injection && error < 0);
+        double rate = activate? gpm->rate(gpmaint_state, current_rate, error, dt) : 0.0;
```

Comment:
I do not understand the "history" comment in the PR description. From what I can tell, rate() is a const function that does not store any history, and the update_gpmaint_target() below also seemingly overwrites a target number, not storing anything. What am I missing?

Also, will the rate() method return a negative number for producers? If not, then the old code was simply buggy, and if it does, the code here is buggy.

Reply:
I see it now, the gpmaint_state is modified.